WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -269,7 +273,23 @@ public SendTabViewModel(WalletViewModel walletViewModel)
 						MainWindowViewModel.Instance.StatusBar.RemoveStatus(dequeuingSelectedCoinsStatusText);
 					}
 
-					BuildTransactionResult result = await Task.Run(() => Global.WalletService.BuildTransaction(Password, new[] { operation }, FeeTarget, allowUnconfirmed: true, allowedInputs: selectedCoinReferences));
+					var result = await Task.Run(() => Global.WalletService.BuildTransaction(Password, new[] { operation }, FeeTarget, allowUnconfirmed: true, allowedInputs: selectedCoinReferences));
+
+					if (IsTransactionBuilder)
+					{
+						var txviewer = IoC.Get<IShell>()?.Documents?.OfType<TransactionViewerViewModel>()?.FirstOrDefault(x => x.Id == Id);
```

Comment:
> IoC.Get<IShell>() cannot be null
ACK: https://github.com/zkSNACKs/WalletWasabi/pull/1351/commits/a791a5cb43a67579491c8e92bf2f0f720cf14fee

> I think Documents cannot be null neither.

Probably. Not sure, I leave it like this for now. I don't want to cause unexpected issues.

> I think we could use the IShell.GetOrCreate here because that method does the same we are doing here.

No, that does something different, however it's good you pointed out, because there was a bug in my code. I had to check the wallet ids not the document ids. Fixed: https://github.com/zkSNACKs/WalletWasabi/pull/1351/commits/a791a5cb43a67579491c8e92bf2f0f720cf14fee